### PR TITLE
 update-event-schema-for-azure-storage-event-grid-notification-for-cr…

### PR DESCRIPTION
**Pull request by wanha-ms**
…eate-delete-directory-and-rename-blob-directory

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (kalyanaj) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: Please revert this to 2018-01-01 as this refers to the API version of EventGrid data plane, and 2018-01-01 is the last known GA version of EventGrid dataplane.

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (kalyanaj) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: Nit: Might want to consider rewording to "The path to the deleted directory."

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (kalyanaj) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: Nit: Might want to reword this to "The new path to the blob after the rename operation".

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (kalyanaj) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: Nit: Might want to reword this to "The new path to the directory after the rename operation".

**Review by kalyanaj — CHANGES_REQUESTED** [review]
I have added my feedback, can you please take a look?

**Comment by wanha-ms** [comment]
Hi Kalyana,

I’ve pushed changes for this PR according to your PR review. Could you please take a look again whenever you had some time?

https://github.com/Azure/azure-rest-api-specs/pull/7509/files

Thank you!

Best Regards
Hao Wang

**Comment by wanha-ms** [comment]
Adding @J. Kalyana Sundaram<mailto:kalyanaj@microsoft.com> explicitly here.
Also adding folks from the team for additional eyes.

Thanks.
________________________________
From: Hao Wang <wanha@microsoft.com>
Sent: Wednesday, October 16, 2019 6:14 PM
To: Azure/azure-rest-api-specs <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Author <author@noreply.github.com>; Ramy Wassef <ramyw@microsoft.com>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)


Hi Kalyana,



I’ve pushed changes for this PR according to your PR review. Could you please take a look again whenever you had some time?



https://github.com/Azure/azure-rest-api-specs/pull/7509/files<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F7509%2Ffiles&data=02%7C01%7Cramyw%40microsoft.com%7C519c0a0c241f4817c99f08d7529f4f80%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637068716492518197&sdata=XxO%2BqZb3esRBhKJMzNrUh0Z7aSxaeD9S6Rnn4UwJuEY%3D&reserved=0>



Thank you!



Best Regards

Hao Wang

**Review by kalyanaj — APPROVED** [review]
Looks good to me.

**Comment by wanha-ms** [comment]
Hi all,

It looks like PR has been approved, all checks has passed. It seems I couldn’t merge the branch to master due to “You’re not authorized<https://help.github.com/articles/about-protected-branches/> to merge this pull request.” Could someone help me out with the merge?

https://github.com/Azure/azure-rest-api-specs/pull/7509

Thanks

Best Regards
Hao Wang

**Comment by wanha-ms** [comment]
I checked the history of this file, but couldn’t figure out who I could ping for the SDK team’s approval. Could someone help me with this?

Thank you

Best Regards
Hao Wang


From: J. Kalyana Sundaram <kalyanaj@microsoft.com>
Sent: Thursday, October 17, 2019 11:36 AM
To: Hao Wang <wanha@microsoft.com>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)

You will need the SDK team’s approval as well and once they approve it they will merge it.

From: Hao Wang <wanha@microsoft.com<mailto:wanha@microsoft.com>>
Sent: Thursday, October 17, 2019 11:14 AM
To: Ramy Wassef <ramyw@microsoft.com<mailto:ramyw@microsoft.com>>; Azure/azure-rest-api-specs <reply@reply.github.com<mailto:reply@reply.github.com>>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com<mailto:azure-rest-api-specs@noreply.github.com>>; J. Kalyana Sundaram <kalyanaj@microsoft.com<mailto:kalyanaj@microsoft.com>>
Cc: Author <author@noreply.github.com<mailto:author@noreply.github.com>>; Vamshi Konagari <Vamshi.Konagari@microsoft.com<mailto:Vamshi.Konagari@microsoft.com>>; Kai Hsu <kaihsu@microsoft.com<mailto:kaihsu@microsoft.com>>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)

Hi all,

It looks like PR has been approved, all checks has passed. It seems I couldn’t merge the branch to master due to “You’re not authorized<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Farticles%2Fabout-protected-branches%2F&data=02%7C01%7Cwanha%40microsoft.com%7C860d0210359f4ed1c85408d75330d34a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069341470911811&sdata=I3nSIZbbOE2oYsQrFDXkYuq9Wh4v5mggP5IZuzpuXtc%3D&reserved=0> to merge this pull request.” Could someone help me out with the merge?

https://github.com/Azure/azure-rest-api-specs/pull/7509<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F7509&data=02%7C01%7Cwanha%40microsoft.com%7C860d0210359f4ed1c85408d75330d34a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069341470921768&sdata=tHUlKj%2Bb9qT7PMiZXxxpF%2FZoYf01NkDqvZ3Op90OVEk%3D&reserved=0>

Thanks

Best Regards
Hao Wang

**Comment by wanha-ms** [comment]
Adding @Fan Qiu<mailto:Fan.Qiu@microsoft.com>  from the SDK team.

Thanks,

Kalyan
From: Hao Wang <wanha@microsoft.com>
Sent: Thursday, October 17, 2019 11:53 AM
To: J. Kalyana Sundaram <kalyanaj@microsoft.com>; Ramy Wassef <ramyw@microsoft.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>; Azure/azure-rest-api-specs <reply@reply.github.com>
Cc: Vamshi Konagari <Vamshi.Konagari@microsoft.com>; Kai Hsu <kaihsu@microsoft.com>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)

I checked the history of this file, but couldn’t figure out who I could ping for the SDK team’s approval. Could someone help me with this?

Thank you

Best Regards
Hao Wang


From: J. Kalyana Sundaram <kalyanaj@microsoft.com<mailto:kalyanaj@microsoft.com>>
Sent: Thursday, October 17, 2019 11:36 AM
To: Hao Wang <wanha@microsoft.com<mailto:wanha@microsoft.com>>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)

You will need the SDK team’s approval as well and once they approve it they will merge it.

From: Hao Wang <wanha@microsoft.com<mailto:wanha@microsoft.com>>
Sent: Thursday, October 17, 2019 11:14 AM
To: Ramy Wassef <ramyw@microsoft.com<mailto:ramyw@microsoft.com>>; Azure/azure-rest-api-specs <reply@reply.github.com<mailto:reply@reply.github.com>>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com<mailto:azure-rest-api-specs@noreply.github.com>>; J. Kalyana Sundaram <kalyanaj@microsoft.com<mailto:kalyanaj@microsoft.com>>
Cc: Author <author@noreply.github.com<mailto:author@noreply.github.com>>; Vamshi Konagari <Vamshi.Konagari@microsoft.com<mailto:Vamshi.Konagari@microsoft.com>>; Kai Hsu <kaihsu@microsoft.com<mailto:kaihsu@microsoft.com>>
Subject: RE: [Azure/azure-rest-api-specs] update-event-schema-for-azure-storage-event-grid-notification-for-cr… (#7509)

Hi all,

It looks like PR has been approved, all checks has passed. It seems I couldn’t merge the branch to master due to “You’re not authorized<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Farticles%2Fabout-protected-branches%2F&data=02%7C01%7Ckalyanaj%40microsoft.com%7C5d58145e34584e54c30608d7533337cf%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069351747209428&sdata=01RtosXXfbV5G8o29OYwnz4gfLg4s1hNYEIpNhIH1JY%3D&reserved=0> to merge this pull request.” Could someone help me out with the merge?

https://github.com/Azure/azure-rest-api-specs/pull/7509<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F7509&data=02%7C01%7Ckalyanaj%40microsoft.com%7C5d58145e34584e54c30608d7533337cf%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637069351747209428&sdata=SL9NwGqlwCiuAbWlS6V9hntZMrUIJ3ogm%2BM%2BSNfVcLY%3D&reserved=0>

Thanks

Best Regards
Hao Wang

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (MyronFanQiu) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: readonly? If some properties are read_only, please add `read_only: true`

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (MyronFanQiu) [inline specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: based on description, should it be a bool type?

**Review by mmyyrroonn — APPROVED** [review]

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (wanha-ms) [reply specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: Will leave this field as it is to keep it consistent with other schemas

**Comment by MyronFanQiu** [comment]
@wanha-ms Hello. Could you fix the errors in LintDiff and Semantic? @NullMDR Hi. Could you help us take a look at SDK's failures?

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (yungezz) [reply specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: HI wanha-ms, what does "consistent with other schemas" mean? read only property need to have read_only: true to make sure downstream sdk generation correctly. It will cause customer failure if swagger not consistent with service behavior.

**Comment by NullMDR** [comment]
@MyronFanQiu The semantic failure should be fixed first to get sdk generation pass.

File: specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json
Comment (wanha-ms) [reply specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json]: This schema is used for EventGrid notifications for changes on Azure Storage. Whenever a file, directory were updated in a storage account, a notification is sent to EventGrid. When I sync with @MyronFanQiu earlier, my thought was, since it is a notification, in this case, all field in this schema is immutable, and should be readonly. However, I didn't see any field is specified in this way in earlier version, or other similar schema definitions. In this way, I think it is better to leave readonly as a future update, instead of in this PR.

**Comment by wanha-ms** [comment]
@MyronFanQiu<mailto:notifications@github.com> @NullMDR<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FNullMDR&data=02%7C01%7Cwanha%40microsoft.com%7Cf2be9fa1c4c64d69583a08d755c89287%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637072192260141647&sdata=ENO2%2FvOlXL8eGupDXXzFldLOUVkN2CBOi6EoU9cp9D0%3D&reserved=0> The SDK build failure has been fixed, after updating “type”: “bool” to “type”: “Boolean”.

There seems to be a mixed usage of “type”: “bool” and “type”: “boolean” in this repo. If someone could help me understand the difference, it would be great.

**Comment by wanha-ms** [comment]
@MyronFanQiu<mailto:notifications@github.com> The build failure has been fixed by updating “type”: “bool” to “type”: “Boolean”.

**Pull request merged**
